### PR TITLE
Properly configure headless run script.

### PR DESCRIPTION
**Pull request by quyykk**
This PR fixes the integration tests that are currently failing due to #6582.

The headless script was configured incorrectly, which this PR fixes by passing appropriate flags to the virtual X server and using `xvfb-run` instead of a custom way to use Xvfb.

### CODE REVIEW

**Comment by quyykk** [comment]
Running the unit tests with various sanitizers also didn't reveal anything. As far as I can understand the problem is as follows: For some reason if we use the `future` to wait, then the integration tests sometimes fail to initialize the software renderer. I can't reproduce on my machine though and so I'm having trouble finding the cause.

**Comment by quyykk** [comment]
After many hours of Googling<sup>tm</sup>, I have finally managed to get it to work.

The problem was how the virtual X server was configured by the headless script. There was like a debug section of the script that I removed because it's not really needed, just run the other script to see the output.

File: source/main.cpp [hunk 1]
```diff
@@ -45,6 +45,7 @@ PARTICULAR PURPOSE.  See the GNU General Public License for more details.
 #include <thread>
 
 #include <cassert>
+#include <future>
```
Comment (petervdmeer) [inline source/main.cpp:48]: Was this statement for debugging (triggering the issue) and should it be removed?

File: tests/run_tests_headless.sh [hunk 2]
```diff
@@ -19,52 +19,13 @@ cd "${HERE}"
...
-
-RETURN_VALUE=$?
-
-kill -s SIGTERM ${XSERVER_PID}
-
-# Enable for debugging:
-# kill -s SIGTERM ${X11VNC_PID}
-
-exit ${RETURN_VALUE}
+xvfb-run --auto-servernum --server-args="+extension GLX +render -noreset" ./run_tests.sh "${EXECUTABLE}" "${RESOURCES}"
```
Comment (petervdmeer) [inline tests/run_tests_headless.sh:30]: That is quite a nice bit shorter than the original code. :)

File: source/main.cpp (hunk 1 above)
Comment (Amazinite) [reply source/main.cpp:48]: This was a missing include from #6582.

**Comment by tehhowch** [comment]
Mind checking whether the fixes to the script also fix integrations on Ubuntu 18.04? Currently they are disabled since integration tests would always fail inexplicably but would work fine for 16.04 and 20.04...

File: tests/run_tests_headless.sh [hunk 3]
```diff
@@ -19,52 +19,13 @@ cd "${HERE}"
 EXECUTABLE="../endless-sky"
 RESOURCES="../"
 
-
-Xvfb :99 -screen 0 1280x1024x24 &
-XSERVER_PID=$!
-if ! ps -p ${XSERVER_PID} > /dev/null 2>&1; then
+if ! command -v xvfb-run &> /dev/null; then
```
Comment (dragonmaus) [inline tests/run_tests_headless.sh:22]: `&>` is a bashism. There's nothing in this script that truly requires bash, so I would recommend switching this back to `> /dev/null 2&>1` (which works on all Bourne shells, including bash), and changing the shebang (line 1) to the more widely available `#!/bin/sh`. The conditionals on line 9 should also be changed from `[[ X == 'Y' ]]` to `[ "X" = 'Y' ]`, for greater compatibility.

File: tests/run_tests_headless.sh [hunk 4]
```diff
@@ -19,52 +19,13 @@ cd "${HERE}"
...
-RETURN_VALUE=$?
-
-kill -s SIGTERM ${XSERVER_PID}
-
-# Enable for debugging:
-# kill -s SIGTERM ${X11VNC_PID}
-
-exit ${RETURN_VALUE}
+xvfb-run --auto-servernum --server-args="+extension GLX +render -noreset" ./run_tests.sh "${EXECUTABLE}" "${RESOURCES}"
+exit $?
```
Comment (dragonmaus) [inline tests/run_tests_headless.sh:31]: This is unnecessary and doubly redundant in any Bourne shell.

`exit` uses the current value of `$?` by default, and the shell effectively and implicitly calls `exit` when it reaches the end of the script.

File: tests/run_tests_headless.sh (hunk 2 above)
Comment (dragonmaus) [inline tests/run_tests_headless.sh:30]: As a micro-optimisation, you can add `exec ` to the start of this line, though this is not necessary by any stretch.

**Review by dragonmaus — CHANGES_REQUESTED** [review]
Some very minor tweaks to improve compatibility.

File: tests/run_tests_headless.sh (hunk 3 above)
Comment (quyykk) [reply tests/run_tests_headless.sh:22]: If I change the conditionals I get `./tests/run_tests_headless.sh: line 9: [: =: unary operator expected`.

**Comment by quyykk** [comment]
@tehhowch Unfortunately the Ubuntu 18 tests still fail. It seems like something is wrong with the drivers there. `glxinfo` correctly reports using the software renderer with OpenGL 3.3, but starting the game fails because the OpenGL context can't be created.

File: tests/run_tests_headless.sh (hunk 3 above)
Comment (quyykk) [reply tests/run_tests_headless.sh]: Scratch that I can't read.

File: tests/run_tests_headless.sh [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
+#!/bin/sh
 set -e
 
 # Run headless "acceptance tests" using Xvfb
 # These tests involve booting the game, loading particular saved games, and making assertions
 # about the state of the game universe. Some tests will involve issuing basic commands.
 
 # TODO: When this script is made cross-platform, replace this OS check with a check for Xvfb and other required resources.
-if [[ $OSTYPE == 'msys' ]] || [[ $OS == 'Windows_NT' ]] || [[ $(uname) == 'Darwin' ]]; then
+if [ "$OSTYPE" == 'msys' ] || [ "$OS" == 'Windows_NT' ] || [ "$(uname)" == 'Darwin' ]; then
```
Comment (dragonmaus) [inline tests/run_tests_headless.sh:9]: Note that you'll want to change the `==` to `=`, as the former doesn't work on some common Bourne shells (such as `dash`).

**Review by dragonmaus — CHANGES_REQUESTED** [review]

File: tests/run_tests_headless.sh (hunk 5 above)
Comment (quyykk) [reply tests/run_tests_headless.sh:9]: First I forget the `""`, then I forgot the `=`. What am I doing. Thanks!

**Review by tehhowch — APPROVED** [review]
Awesome!

**Pull request merged**